tensorrt_llm/_torch/pyexecutor/resource_manager.py

```
@@ -1627,6 +1642,300 @@ def reset_reuse_state(self):
+        # Each sub-manager needs blocks for its own layer group; use the max.
+        return max(m.get_needed_resource_to_completion(request)
+                   for m in self.sub_managers)
+
+    def prepare_resources(self, scheduled_batch: ScheduledRequests):
+        for mgr in self.sub_managers:
+            mgr.prepare_resources(scheduled_batch)
+
+    def update_resources(self,
+                         scheduled_batch: ScheduledRequests,
+                         attn_metadata: "AttentionMetadata" = None,
+                         kv_cache_dtype_byte_size: float = None):
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)